src/domains/queries.ml

```diff
@@ -312,6 +321,11 @@ struct
       | Any (Invariant i1), Any (Invariant i2) -> compare_invariant_context i1 i2
       | Any (InvariantGlobal vi1), Any (InvariantGlobal vi2) -> Stdlib.compare (Hashtbl.hash vi1) (Hashtbl.hash vi2)
       | Any (IterSysVars (vq1, vf1)), Any (IterSysVars (vq2, vf2)) -> VarQuery.compare vq1 vq2 (* not comparing fs *)
+      | Any (MutexType (v1,o1)), Any (MutexType (v2,o2)) -> let r = CilType.Varinfo.compare v1 v2 in
+        if r <> 0 then
+          r
+        else
+          Lval.OffsetNoIdx.compare o1 o2
```

Comment:
As mentioned above, ideally this would just use an already existing type for mutex lvals, so this comparison wouldn't have to be part of queries themselves. Either way, `[%ord: CilType.Varinfo.t * Lval.OffsetNoIdx.t]` is how to just derive these definitions inline.

Reply:
I derive them now.